soroban-env-host/src/host/metered_map.rs

```diff
@@ -84,6 +84,9 @@ where
     }
 
     pub fn from_map(map: Vec<(K, V)>, ctx: &Ctx) -> Result<Self, HostError> {
+        if u32::try_from(map.len()).is_err() {
```

Comment:
No strong opinion here, just a thought.: while I understand this check in theory, it seems to only introduce some additional confusion. There shouldn't ever be objects that are even close to 100 million elements, not to mention billions. So overflow check seems somewhat arbitrary and confusing to me here as it implies that it's possible to create a big enough vector somehow.

Reply:
Agreed, it's strictly a theoretical condition. The testcase submitted to demonstrate it had the budget turned off (and when I ran it it consumed 45GiB RAM on my workstation). Also it hinges on usize not being u64, which in practice is never will be either (nobody runs 32-bit hosts anymore). This is a dotting "i"s, crossing "t"s issue. A formality.